src/main/service/dataManipulationService.ts

```diff
@@ -116,6 +116,75 @@ export class DataManipulationService {
     return sscsDailyListData;
   }
 
+  /**
+   * Reshaping etDailyList json data into formatted niceness.
+   */
+  public reshapeEtDailyListData(etDailyList: string): object {
+    const etDailyListData = JSON.parse(etDailyList);
+    let hearingCount = 0;
+    etDailyListData['courtLists'].forEach(courtList => {
+      courtList['courtHouse']['courtRoom'].forEach(courtRoom => {
+        courtRoom['session'].forEach(session => {
+          session['formattedJudiciary'] = this.getJudiciaryNameSurname(session);
+          delete session['judiciary'];
+          session['sittings'].forEach(sitting => {
+            hearingCount = hearingCount + sitting['hearing'].length;
+            sitting['sittingStartFormatted'] = this.publicationTimeInBst(sitting['sittingStart']);
+            this.calculateDuration(sitting);
+            this.findAndConcatenateHearingPlatform(sitting, session);
+            sitting['hearing'].forEach(hearing => {
+              this.findAndManipulateClaimantsForEtLists(hearing);
+            });
+          });
+        });
+        courtRoom['totalHearing'] = hearingCount;
+        hearingCount = 0;
+      });
+    });
+    return etDailyListData;
+  }
+
+  private findAndManipulateClaimantsForEtLists(hearing: any): void {
```

Comment:
I have CLAIMANT and CLAIMANT_PETITIONER for IAC daily list too, and have a different way to manipulate the party information. So will have merging conflicts later

Reply:
Should work nice now - cheers for the chat earlier :)